doc/README.notifications

```diff
@@ -0,0 +1,37 @@
+During the life-cycle of a user-space filesystem the usual flow is:
+
+  1. User-space application does a filesystem-related syscall
+  2. Kernel VFS calls into the FUSE kernel driver
+  3. FUSE kernel redirects request to the user-space filesystem
+  4. User-space server replies to request
+  5. FUSE returns reply to VFS
+  6. User-space application gets reply from the kernel
+
+However, there are occasions where the filesystem needs to send notifications to
+the kernel that are not in reply to any particular request.  If, for example,
+when a READ request of 4096 bytes results in the filesystem having more data
+available for the specific inode, it may be useful to provide this extra data to
+the kernel cache so that future read operations will be faster.
```

Comment:
Sounds a bit awkward for me, but I'm not a native speaker. The AI system at work suggests

Augment
When a filesystem has more data available than requested for an inode, providing this extra data to the kernel cache can optimize subsequent read operations.